base_conditional_image/models/abstract_image.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from openerp import fields, models
+from openerp.tools.safe_eval import safe_eval
+
+
+class AbstractImageStorage(models.AbstractModel):
+    _name = 'abstract.conditional.image'
+
+    image = fields.Binary(
+        compute='_compute_image', string="Image",
+        store=False, readonly=True
+    )
+    image_medium = fields.Binary(
+        compute='_compute_image', string="Image",
+        store=False, readonly=True
+    )
+    image_small = fields.Binary(
+        compute='_compute_image', string="Image",
+        store=False, readonly=True
+    )
+
+    def _compute_image(self):
+        if 'company_id' in self._fields:
+            images = self.env['image'].search(
+                [('model_name', '=', cls._name)],
```

Comment:
`cls` is not defined, you can use `self`.

Reply:
You're right, I missed it during refactoring